src/upload/form/project_assets.rs

```diff
@@ -73,3 +73,94 @@ impl ProjectAssets {
         self.script_path.clone()
     }
 }
+
+pub struct Module {
+    pub path: PathBuf,
+    pub module_type: ModuleType,
+}
+
+impl Module {
+    pub fn new(path: PathBuf) -> Result<Module, failure::Error> {
+        let extension = path
+            .extension()
```

Comment:
This will fail on a file like `.gitignore`.

Reply:
I'm not sure what we should do for invalid files in build_dir... personally I'd rather fail hard on encountering one, and require users to have their build output a clean directory, rather than support ignore logic

Reply:
updating the error message here to print the file name so the error is more actionable for the user